nvme_call_home: remove -d and -f as visible options

gjoyce-ibm opened the pull request:
The dump(-d) and file(-f) options have been removed as option in usage and also in the man page. The options are for internal test only and should not have been exposed. Also added some helpful status messages.

Kyle-Mahlkuch reviewed (commented):
I reviewed this commit, it looks correct.

sathvika-v reviewed (commented):
Hi @gjoyce-ibm, Thanks for the PR.

gjoyce-ibm commented:
> Hi @gjoyce-ibm, Thanks for the PR. Listing down a few of my thoughts below..
> 
> A more descriptive message indicating success/failure of the command may be helpful. Something like "NVMe device health status diagnosed successfully".
> 
> I think it's also better to add an error message if read_file_dict() function fails. Currently, we are only doing return -1. Probably a message indicating failure to read a file and parse values would also help.
> 
> The commit description is better to be up to 72 characters.
> 
> Thanks, Sathvika

I updated the success/failure messages slightly. Since the command may not diagnose health status I didn't include that. The message will generally not be seen by a user as the standard use case is running it from a cron job.

Added an error message if read_file_dict() fails. There are also existing error messages in that function.

I'm not quite clear about the commit description. I keep the initial line to 55 chars or less and subsequent lines to 70 chars or less as this conforms to kernel standards. Could you elaborate on your comment?

sathvika-v commented:
> I'm not quite clear about the commit description. I keep the initial line to 55 chars or less and subsequent lines to 70 chars or less as this conforms to kernel standards. Could you elaborate on your comment?

Yes, I usually keep it to 72 chars or less. Current commit message follows 72 char limit. Last time when I checked, it was slightly over 72 chars, or I may have miscalculated it. 

The changes look good to me. It would be better to add a one-line commit description after "diag_nvme: improvements to status and err messages" commit.

Reviewed-by: Sathvika Vasireddy <sv@linux.ibm.com>

maheshsal commented on diags/diag_nvme.c:
```diff
@@ -974,11 +988,6 @@ extern int open_nvme(char *dev_path) {
 static void print_usage(char *command) {
 	printf("Usage: %s [-h] [-d <file>] [-f <file>] [<nvme_devices>]\n"
```
The print_usage still prints [-d <file>] [-f <file>] . Should those also be removed ?

gjoyce-ibm replied:
Yes, good catch. They were removed from the details but not the core usage on the first line of the message.
